src/lv_objx/lv_kb.c

```diff
@@ -250,6 +253,18 @@ void lv_kb_set_style(lv_obj_t * kb, lv_kb_style_t type, const lv_style_t * style
     }
 }
 
+void lv_kb_set_map(lv_obj_t * kb, lv_kb_shift_t shift, const char * map[])
+{
+		kb_map[shift] = map;
+		lv_btnm_set_map(kb, map);
```

Comment:
I think it would be better to separate changing the keyboard's state and changing the button map for that state.

Reply:
In lv_kb_set_map() I needed to use lv_btnm_set_map() otherwise the default button map does not get updated.

Calling lv_kb_set_mode() after lv_kb_set_map() checks for a mode difference and so does not update the maps.

Making ext->mode invalid would allow lv_kb_set_mode() to update the maps but is a kludge.

Calling lv_kb_set_mode() twice after lv_kb_set_map() would also work but is also a kludge.

The problem is that lv_kb_def_event_cb() changes the keyboard map without storing the shift state.

I can store the shift state but think it might conflict with dev-7.0.  I'll look now.

Reply:
Adding lv_kb_shift_t to the ext object and a helper function to update the maps works fine.  

My file merge and diff are really messy and I'm not sure if the changes will go into 7.0.

Now the next thing is to get rid of the hard coded text 'abc', 'ABC' & '1#' for shift state changes to allow symbols.

Reply:
> My file merge and diff are really messy and I'm not sure if the changes will go into 7.0.

The conflicts appear to be very minor. It will be easy for me to fix them when we are ready to merge this.

Reply:
Is it better to PR on the dev branch rather than the production[?] code?

Reply:
> Is it better to PR on the dev branch rather than the production[?] code?

In general, yes. PRs should only be sent to `master` if they are bugfixes or documentation improvements.

Otherwise, non-API breaking changes should go to the dev branch that will be released first and is not feature frozen. Right now, that would be `dev-7.0`, as `dev-6.1` has been feature-frozen.

API-breaking changes have to wait for the next major version to get a development branch created. Then they can be sent to that branch.

Hopefully that makes sense.